backend/test/models/user.js

```diff
@@ -0,0 +1,156 @@
+                    type: 'Point',
+                    coordinates: [1.234, 5.678]
+                }
+            };
+
+            const result = validate(user);
+            chai.expect(result.error.details[0].message).to.be.equal('"name" is required');
+        });
+
+        it('should throw an error when email is missing', () => {
+            const user = {
+                name: 'Test User',
+                phone: '1234567890',
+                password: 'password123',
+                userType: 'customer',
+                location: {
+                    type: 'Point',
+                    coordinates: [1.234, 5.678]
+                }
+            };
+
+            const result = validate(user);
+            chai.expect(result.error.details[0].message).to.be.equal('"email" is required');
+        });
+
+        it('should throw an error when password is missing', () => {
+            const user = {
+                name: 'Test User',
+                email: 'test@example.com',
+                phone: '1234567890',
+                userType: 'customer',
+                location: {
+                    type: 'Point',
+                    coordinates: [1.234, 5.678]
+                }
+            };
+
+            const result = validate(user);
+            chai.expect(result.error.details[0].message).to.be.equal('"password" is required');
+        });
+
+        it('should throw an error when phone is missing', () => {
+            const user = {
+                name: 'Test User',
+                email: 'test@example.com',
+                password: 'password123',
+                userType: 'customer',
+                location: {
+                    type: 'Point',
+                    coordinates: [1.234, 5.678]
+                }
+            };
+
+            const result = validate(user);
+            chai.expect(result.error.details[0].message).to.be.equal('"phone" is required');
+        });
+
+        it('should throw an error when location is missing', () => {
+            const user = {
+                name: 'Test User',
+                email: 'test@example.com',
+                phone: '1234567890',
+                password: 'password123',
+                userType: 'customer',
+            };
+
+            const result = validate(user);
+            chai.expect(result.error.details[0].message).to.be.equal('"location" is required');
+        });
+
+        it('should throw an error when location coordinates are missing', () => {
+            const user = {
+                name: 'Test User',
+                email: 'test@example.com',
+                phone: '1234567890',
+                password: 'password123',
+                userType: 'customer',
+                location: {
+                    type: 'Point'
+                }
+            };
+
+            const result = validate(user);
+            chai.expect(result.error.details[0].message).to.be.equal('"coordinates" is required');
```

Comment:
Because Co-ordinates is under the location object it should be `location.coordinates is required`.

```js
      chai.expect(result.error.details[0].message).to.be.equal('"location.coordinates" is required')

```